backend/internal/system/healthcheck/service/healthcheckservice.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package service provides health check-related business logic and operations.
+package service
+
+import (
+	dbmodel "github.com/asgardeo/thunder/internal/system/database/model"
+	"github.com/asgardeo/thunder/internal/system/database/provider"
+	"github.com/asgardeo/thunder/internal/system/healthcheck/model"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+// HealthCheckServiceInterface defines the interface for the health check service.
+type HealthCheckServiceInterface interface {
+	CheckReadiness() model.ServerStatus
+}
+
+// HealthCheckService is the default implementation of the HealthCheckServiceInterface.
+type HealthCheckService struct{}
+
+// GetHealthCheckService creates a new instance of HealthCheckService.
+func GetHealthCheckService() HealthCheckServiceInterface {
+	return &HealthCheckService{}
+}
+
+// CheckReadiness checks the readiness of the server and its dependencies.
+func (as *HealthCheckService) CheckReadiness() model.ServerStatus {
+	configDBStatus := model.ServiceStatus{
+		ServiceName: "IdentityDB",
+		Status:      checkDatabaseStatus("identity", queryConfigDBTable),
+	}
+
+	runtimeDBStatus := model.ServiceStatus{
```

Comment:
Wondering whether we should make checking DB status configurable. In a cloud deployment there maybe seperate checks for DB health and checking it again from the server maybe redudant.

Reply:
When you say "checking DB status configurable", are you suggesting to perform the check based on its enabled or disabled?

IMO, readiness service should indicate the availability/reachability of the its dependent services. How it get that info can be varied. We need to to have the flexibility on how we perform that check.. But we can bring that full support in a another iteration.